lib/src/main/kotlin/com/lemonappdev/konsist/core/util/TypeUtil.kt

```diff
@@ -93,24 +94,27 @@ object TypeUtil {
                 .firstOrNull { import -> import.name.substringAfterLast(".") == typeText }
                 ?.name
 
-        val declarations =
+        val declarationFqn =
             containingFile
                 .declarations()
-                .filterIsInstance<KoFullyQualifiedNameProvider>()
-                .filter { it.fullyQualifiedName?.endsWith(typeText ?: "") == true }
-
-        val parentDeclFqn = (parentDeclaration as? KoFullyQualifiedNameProvider)?.fullyQualifiedName.orEmpty()
-
-        val decl =
-            declarations.singleOrNull()
-                ?: declarations.firstOrNull { decl ->
-                    decl.fullyQualifiedName?.contains(parentDeclFqn) == true ||
-                        ((decl as? KoContainingDeclarationProvider)?.containingDeclaration as? KoDeclarationProvider)?.hasDeclaration {
-                            it == parentDeclaration
-                        } == true
-                }
-
-        fqn = fqn ?: decl?.fullyQualifiedName
+                .getDeclarationFullyQualifiedName(typeText, parentDeclaration)
+
+        fqn = fqn ?: declarationFqn
```

Comment:
1. `declarationFqn` should be only retrieved if `fqn` (rename to `fullyQualifiedName`) is null (code will be more optimized)
2. Rename `declarationFqn` to `declarationFullyQualifiedName`n`